feat: complete events.Event docs (#35360)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/events/event/removerules/index.md

---
title: Event.removeRules()
slug: Mozilla/Add-ons/WebExtensions/API/events/Event/removeRules
page-type: webextension-api-event
browser-compat: webextensions.api.events.Event.removeRules
---

{{AddonSidebar}}

> Removes an array of rules that were previously registered to the event.

Reviewer: ```suggestion
Removes rules that were registered on a declarative event using {{WebExtAPIRef("events.Event.addRules()")}}.
```

## Syntax

```js-nolint
events.Event.removeRules(ruleIdentifiers, callback)
```

### Parameters

- `ruleIdentifiers` {{optional_inline}}

>   - : `array` of `string`. If an array of `string` is passed, only rules with identifiers contained in this array are removed, otherwise all rules registered are removed.

Reviewer: ```suggestion
  - : `array` of `string`. Identifiers for the rules to remove. If no array is passed, all rules are removed.
```
